packages/eslint-plugin/tests/rules/no-base-to-string.test.ts

```diff
@@ -483,6 +483,28 @@ function foo<T>(x: T) {
 declare const u: unknown;
 String(u);
     `,
+    `
+type Value = string | Value[];
+declare const v: Value;
+
+String(v);
+    `,
+    `
+type Value = (string | Value)[];
+declare const v: Value;
+
+String(v);
+    `,
+    `
+type Value = [Value, Value];
```

Comment:
To my understanding, types like this should be valid to stringify (also similar ones like `type Value = Value[]`).